pkg/resources/management.cattle.io/v3/globalrole/validator.go

```diff
@@ -232,27 +245,34 @@ func (a *admitter) validateUpdateFields(oldRole, newRole *v3.GlobalRole, fldPath
 	return nil
 }
 
-// isMetaOnlyChange checks if old and new are deep equal in all fields except metadata and typemeta. Will return false on a
-// non-effectual change.
-func isMetaOnlyChange(oldGR *v3.GlobalRole, newGR *v3.GlobalRole) bool {
-	// if the metadata between old/new hasn't changed, then this isn't a metadata only change
-	if grMetaEqual(oldGR, newGR) {
-		return false
+func (a *admitter) hasEscalateVerb(request *admission.Request, name string, namespace string) (bool, error) {
+	extras := map[string]v1.ExtraValue{}
+	for k, v := range request.UserInfo.Extra {
+		extras[k] = v1.ExtraValue(v)
 	}
 
-	return grContentEqual(oldGR, newGR)
-}
-
-func grMetaEqual(oldGR *v3.GlobalRole, newGR *v3.GlobalRole) bool {
-	if oldGR == newGR {
-		// same pointer or both objects are nil
-		return true
-	}
-	if oldGR == nil || newGR == nil {
-		// one object is nil and the other is not.
-		return false
+	resp, err := a.sar.Create(request.Context, &v1.SubjectAccessReview{
+		Spec: v1.SubjectAccessReviewSpec{
+			ResourceAttributes: &v1.ResourceAttributes{
+				Verb:      "escalate",
+				Namespace: namespace,
+				Version:   gvr.Version,
+				Resource:  gvr.Resource,
+				Group:     gvr.Group,
+				Name:      name,
+			},
+			User:   request.UserInfo.Username,
+			Groups: request.UserInfo.Groups,
+			Extra:  extras,
+			UID:    request.UserInfo.UID,
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return false, fmt.Errorf("failed to checkout create sar request: %w", err)
 	}
-	return reflect.DeepEqual(oldGR.ObjectMeta, newGR.ObjectMeta)
+
+	return resp.Status.Allowed, nil
+
 }
 
 // grContentEqual checks if two globalRoles have equivalent content (all fields excluding metadata && typeMeta).
```

Comment:
remove gr content equal and replace with reflect.DeepEquals(). Ignore delta to object meta, type meta, and new user default.